### ISSUE TITLE
Particle issues on Vulkan renderer (O3DE Atom)

### ISSUE THREAD

**Issue by AMZN-Olex** [PopcornFX/O3DEPopcornFXPlugin#36]
A ribbon trail particle looks fine with DX12 Atom:
![image](https://user-images.githubusercontent.com/5432499/221319674-6354beb7-686a-485b-8fea-7b2e49ccba3b.png)

However, the same level and the same particle effect looks wrong with a Vulkan renderer:
![image](https://user-images.githubusercontent.com/5432499/221319718-78ccc1ab-38e4-4059-84d4-a079b1ca61dc.png)

For reference the effect is "Straight_Shot_Effect.pkfx" and I'm attacking a zip of the PopcornFX project.
[PopcornFX.zip](https://github.com/PopcornFX/O3DEPopcornFXPlugin/files/10829698/PopcornFX.zip)

'-rhi=vulkan -rhi-device-validation=enable':
[Game_vulkan.log](https://github.com/PopcornFX/O3DEPopcornFXPlugin/files/10829757/Game_vulkan.log)

'-rhi=dx12 -rhi-device-validation=enable':
[Game_dx12.log](https://github.com/PopcornFX/O3DEPopcornFXPlugin/files/10829759/Game_dx12.log)

**Comment by AMZN-Olex** [comment]
Note, the switch can be done from a command line: `-rhi=vulkan` or `-rhi=dx12`

**Comment by moudgils** [comment]
Can you attach output log after enabling device validation? It may provide further leads as to where the issue maybe.

**Comment by AMZN-Olex** [comment]
> Can you attach output log after enabling device validation? It may provide further leads as to where the issue maybe.

@moudgils Done.

**Comment by ValPKFX** [comment]
Thank you, I will try to reproduce it and see if we can do something.

**Comment by moudgils** [comment]
![image (7)](https://user-images.githubusercontent.com/47460854/223195987-bb39c970-243a-4584-baa4-1d6174015ba9.png)
Took a quick look at the particle effect issue. The problem is with the vertex data for the particle effect in questions. For Vulkan it looks messed up. Vk is to the left and dx12 is to the right. If I had to take a guess its the maybe the sim code didnt update the gpu memory correctly.

**Comment by lmbr-pip** [comment]
@ValPKFX Any updates here?

**Comment by lmbr-pip** [comment]
FYI We just made o3de-multiplayer-sample, and o3de-multiplayersample-assets all public. If you build latest MPS (would recommend using the stabilization/2305 branches), you can see this just in the Editor viewport without needing to run actual game. Impacts several VFXs in this project.

**Comment by HugoPKFX** [comment]
Hi, thanks for reporting the issue.
Val is out this week unfortunately, I'll try to take a look.
GPU Buffers fill logic isn't API dependent in the plugin / PopcornFX runtime, things that come to mind to double check:
* API specific buffer map/unmap differences
* Multi buffering issues
* Buffer creation flags

**Comment by HugoPKFX** [comment]
Resolved on development branch, commit id `92ea37adb2ad0ee94d4bfd3d4ea2dbbd0220094f`
Also fixed in upcoming PopcornFX v2.15.5
Thanks for reporting the issue

Issue came from Atom RHI difference between D3D12 and Vulkan. For Ribbon particles, we were incorrectly doing the vertex input positions setup (via `AZ::RHI::InputStreamLayoutBuilder`):

```cpp
// Before:
layoutBuilder.AddBuffer()->Channel("POSITION", AZ::RHI::Format::R32G32B32_FLOAT);
// After:
layoutBuilder.AddBuffer()->Channel("POSITION", AZ::RHI::Format::R32G32B32A32_FLOAT);
```
This second parameter is ignored in D3D12 Atom/RHI, as vertex buffer stride is specifed when `IASetVertexBuffers` is called.
In Vulkan byte stride is specified when the graphics pipeline is created, ignoring the `RHI::StreamBufferView` infos.

Additional debug asserts in Atom or errors could help quickly identify such issues

**Comment by ValPKFX** [comment]
There was no PR, it's the commit 92ea37adb2ad0ee94d4bfd3d4ea2dbbd0220094f
